cli/state.rs

```diff
@@ -174,13 +160,14 @@ impl Loader for ThreadSafeState {
   ) -> Box<deno::SourceCodeInfoFuture> {
     self.metrics.resolve_count.fetch_add(1, Ordering::SeqCst);
     Box::new(
-      fetch_module_meta_data_and_maybe_compile_async(self, module_specifier)
-        .map(|module_meta_data| deno::SourceCodeInfo {
-          // Real module name, might be different from initial URL
+      fetch_source_file_and_maybe_compile_async(self, module_specifier).map(
+        |source_file| deno::SourceCodeInfo {
```

Comment:
I think we should return `CompiledModule` from `fetch_source_file_and_maybe_compile_async`:
```
struct CompiledModule {
   name: String,
   code: String,
}
```

It'd be made from:
- original source file for JS and Unknown media types
- original source file wrapped in export for JSON media type
- output code of compiled JS file for TypeScript media type